### ISSUE TITLE
Bug: Array of type object not validating 

### ISSUE THREAD

**Issue by jdmeyer3** [StackStorm/st2web#710]
If an action has the following for a parameter type in an action:
```yaml
parameters:
  test_array_obj:
    type: array
    items:
      type: object
    description: "array of objects to evaluate"
    required: true
    position: 0
```
There is an issue with the UI where I cannot enter an array of objects into the parameter field.
![image](https://user-images.githubusercontent.com/25443446/57704218-b6d4b300-761e-11e9-94c3-9f81b58cba59.png)

I've tried many permutations on how to enter the array of objects, but the UI cannot evaluate it correctly.

## Installed Versions
st2.x86_64                         2.10.1-1               @StackStorm_stable
st2-auth-ldap.x86_64               2.10.4-1               @StackStorm_enterprise
st2chatops.x86_64                  2.10.1-1               @StackStorm_stable
st2flow.x86_64                     2.10.4-1               @StackStorm_enterprise
st2mistral.x86_64                  2.10.1-1               @StackStorm_stable

## Distro 
CentOS 7

**Comment by MichaelMcClure** [comment]
This might be relevant:  https://github.com/StackStorm/st2web/issues/394

**Comment by LindsayHill** [comment]
Resolved by #711

**Comment by VineeshJain** [comment]
Reopening the issue as the proposed fix caused breakage in array type fields. Please refer to #711 for further details.
